docs/modules/ROOT/pages/development.adoc

```diff
@@ -0,0 +1,24 @@
+= Development
+
+This page describes considerations when developing RSpec-specific cops. It is intended to be a complement to the general https://docs.rubocop.org/rubocop/development.html[RuboCop development documentation].
+
+== Base class
+
+The `RuboCop::Cop::RSpec::Base` class includes convenient https://docs.rubocop.org/rubocop-ast/node_pattern.html[node pattern DSL] matchers that will automatically account for any xref:usage.adoc#rspec-dsl-configuration[custom RSpec DSL configuration].
+
+For example, if the project defines https://github.com/test-prof/test-prof/blob/master/docs/recipes/let_it_be.md[`let_it_be`] as a `Helper`, then all cops will find `let_it_be` when using the `let?` matcher.
+
+== Writing specs
+
+When working on RSpec-specific cops, ensure that the https://github.com/rubocop/rubocop-rspec/blob/master/config/default.yml[default language config] is loaded for all RSpec specs:
+
+[source,ruby]
+----
+require 'rubocop/rspec/shared_contexts/default_rspec_language_config_context'
+
+RSpec.config do |config|
+  # ...
+  # Add metadata as appropriate
+  config.include_context 'with default RSpec/Language config'
+end
```

Comment:
```ruby
  # Set metadata on all cop specs
  config.define_derived_metadata(file_path: %r{/spec/rubocop/cop/}) do |meta|
    meta[:type] = :cop_spec
  end

  # Include config shared context for all cop specs
  config.define_derived_metadata(type: :cop_spec) do |meta|
    meta[:config] = true
  end

  config.include_context 'with default RSpec/Language config', :config
```

Notice the difference that the context that defines `config` will only be included in specs from certain directories, or explicitly marked as `:cop_spec`.

This is only important for monolithic projects that also define cops among all, not to litter **all** specs with the populated `config`.
For custom RuboCop extensions that inspect specs, your current approach would work perfectly fine.

Reply:
@pirj this is what I was getting at with 'add metadata as appropriate', but I agree it's better to be clear here. I've added this change and a note that it's an example above the code block.